pkg/ibmvpc/parser.go

```diff
@@ -632,6 +659,58 @@ func getNACLconfig(rc *ResourcesContainer, res *vpcmodel.CloudConfig, subnetName
 	return nil
 }
 
+func getSubnetByIPAddress(address string, c *vpcmodel.CloudConfig) (subnet *Subnet, err error) {
+	addressIPblock := common.NewIPBlockFromCidrOrAddress(address)
+	for _, s := range c.NodeSets {
+		if s.Kind() == ResourceTypeSubnet {
+			subnetRange := s.AddressRange()
+			if addressIPblock.ContainedIn(subnetRange) {
+				return s.(*Subnet), nil
+			}
+		}
+	}
+	return nil, fmt.Errorf("could not find matching subnet for address %s", address)
+}
+
+func getVPEconfig(rc *ResourcesContainer, res *vpcmodel.CloudConfig) (err error) {
+	for i := range rc.vpeList {
+		vpe := rc.vpeList[i]
+		vpc, err := getVPCObjectByName(res, *vpe.VPC.Name)
+		if err != nil {
+			return err
+		}
+		vpeResource := &Vpe{
+			VPCResource: vpcmodel.VPCResource{ResourceName: *vpe.Name, ResourceUID: *vpe.CRN, ResourceType: ResourceTypeVPE},
+			vpc:         vpc,
+		}
+		res.NodeSets = append(res.NodeSets, vpeResource)
+		rIPList := vpe.Ips // reserved ips bound to this endpoint gateway
+		for _, rIP := range rIPList {
+			rIPNode := &ReservedIP{
+				VPCResource: vpcmodel.VPCResource{ResourceName: *rIP.Name, ResourceUID: *rIP.ID,
+					ResourceType: ResourceTypeReservedIP, Zone: ""}, // the zone gets updated later
+				vpe:     *vpe.Name,
+				address: *rIP.Address,
+			}
+			subnet, err := getSubnetByIPAddress(*rIP.Address, res)
+			if err != nil {
+				return err
+			}
+			rIPNode.subnet = subnet
+			rIPNode.Zone = subnet.ZoneName()
+			res.Nodes = append(res.Nodes, rIPNode)
+			// TODO: make sure the address is in the subnet's reserved ips list?
+			subnet.nodes = append(subnet.nodes, rIPNode)
+			res.NameToResource[rIPNode.Name()] = rIPNode
+			vpeResource.nodes = append(vpeResource.nodes, rIPNode)
+		}
+		res.NameToResource[vpeResource.ResourceName] = vpeResource
+		// TODO: verify that vpe.SecurityGroups contain the reserved-ips as members? (not at this stage)
+		// sgList := vpe.SecurityGroups
```

Comment:
Is this TODO still relevant?

Reply:
Yes, because currently there is no validation that all the reserved IPs of the vpe are indeed members of its security groups, the parser only takes the security groups and sets the correct references to the specified members.